libs/platform/src/lib/components/dynamic-page/dynamic-page.component.ts

```diff
@@ -266,24 +301,39 @@ export class DynamicPageComponent extends BaseComponent implements AfterContentI
      * set styles for tab labels
      */
     private _setTabStyles(): void {
-        const tabList = this._elementRef.nativeElement.querySelector('.fd-tabs');
-        if (tabList) {
-            this._addClassNameToCustomElement(tabList, CLASS_NAME.dynamicPageTabs);
-            this._addClassNameToCustomElement(tabList, CLASS_NAME.dynamicPageTabsAddShadow);
-            if (this.size) {
-                this._setTabsSize(this.size, tabList);
-            }
-            if (this.headerComponent?.collapsible) {
-                const pinCollapseShadowElement = this._elementRef.nativeElement.querySelector(
-                    '.fd-dynamic-page__collapsible-header-visibility-container'
-                );
-                if (pinCollapseShadowElement) {
-                    this._addClassNameToCustomElement(
-                        pinCollapseShadowElement,
-                        CLASS_NAME.dynamicPageCollapsibleHeaderPinCollapseNoShadow
-                    );
-                }
-            }
+        if (!this.contentContainer) {
+            return;
+        }
+        const tabList: HTMLElement = this.contentContainer.nativeElement.querySelector('.fd-tabs');
+        if (!tabList) {
+            return;
+        }
+
+        this._addClassNameToCustomElement(tabList, CLASS_NAME.dynamicPageTabs);
+        this._addClassNameToCustomElement(tabList, CLASS_NAME.dynamicPageTabsAddShadow);
+        this._renderer.setStyle(tabList, 'z-index', 1);
+
+        if (this.header) {
+            this._renderer.setStyle(this.header.nativeElement, 'z-index', 2);
+        }
+
+        if (this.size) {
+            this._setTabsSize(this.size, tabList);
+        }
+
+        if (!this.headerComponent?.collapsible) {
+            return;
+        }
+
+        const pinCollapseShadowElement = this._elementRef.nativeElement.querySelector(
+            '.fd-dynamic-page__collapsible-header-visibility-container'
+        );
+
+        if (pinCollapseShadowElement) {
+            this._addClassNameToCustomElement(
+                pinCollapseShadowElement,
+                CLASS_NAME.dynamicPageCollapsibleHeaderPinCollapseNoShadow
+            );
         }
```

Comment:
Let's recheck if this is still needed, when applied static/relative position

Reply:
this is still needed as it is unrelated to positioning. It adds some necessary modifier classes to the tabs.